src/OrchardCore.Modules/OrchardCore.Taxonomies/Drivers/TaxonomyFieldDriver.cs

```diff
@@ -32,95 +33,76 @@ public class TaxonomyFieldDisplayDriver : ContentFieldDisplayDriver<TaxonomyFiel
 
         public override IDisplayResult Display(TaxonomyField field, BuildFieldDisplayContext context)
         {
-            return Initialize<DisplayTaxonomyFieldViewModel>("TaxonomyField", model =>
+            if (String.IsNullOrEmpty(context.PartFieldDefinition.DisplayMode()))
```

Comment:
I faced the same issue when I tried to implement a custom editor. I would not want the default driver for the field to be called. I am wondering if this should be a default behavior, because someone might have the same issue with other fields, if the editors don't match exactly the driver. OR we should have a way to express in the editor/display that it should not trigger the default driver.

Reply:
I think the best way would be to register the drivers for each editor we want. So when we create a new editor, we also register which drivers we want for this editor. We can register the existing one, or a new one. And someone could register its driver for "any/*" editor.

Something like an option (like it was done for fields/parts) where we can set which driver to use to each display mode. Probably on this exact option object.

Reply:
> Something like an option (like it was done for fields/parts) where we can set which driver to use to each display mode. Probably on this exact option object.

Funny you should mention that. I have a pr here https://github.com/OrchardCMS/OrchardCore/pull/4361 which registers display drivers and handlers through options. 
It could very easily be extended to support registering against a display mode & editor mode.

If you are able to review that PR, then I would look to extending it to support registration and a lookup based on `DisplayMode` and `Editor`